### DISCUSSION TITLE
contact form/next js

### DISCUSSION THREAD

**Discussion by PhoenixDevStudio** [vercel/next.js#26048] [Help]
Can we create a back-end contact form for nextjs, that uses a free domain that we don't have to spend money on or they any alternative where I can just send the info to my email(gmail) directly without the domain.

**Comment by michrome** [comment]
You could use something like [Supabase](https://supabase.io/) to receive the form submission. It has a free plan.

**Reply by PhoenixDevStudio** [reply]
Is it more recommended than netifly, graphQl or even Vercel

**Reply by michrome** [reply]
They’re used in conjunction with each other. You could host your Next.js app on Vercel, and use GraphQL to read/write data from Supabase.

**Comment by juanbzz** [comment]
This is what I'd do:

* Create a function to send an email using your Gmail account with something like [nodemailer](https://www.npmjs.com/package/nodemailer)
* Create an [API Route](https://nextjs.org/docs/api-routes/introduction) named `send-email.js` to send an email with data from a request like `{ email: "john@email.com", message: "Hi" }`
* Make sure this API Route can be used only by your website, otherwise it will get abused.
* Use the API anywhere in your site sending `POST` requests with `fetch` or `axios` if you prefer.
